### ISSUE TITLE
Implement status-rich incremental updates for complex objects

### ISSUE THREAD

**Issue by hausdorff** [pulumi/pulumi-kubernetes#93]
Specifically for the following API objects:

- [x] `Service`
- [x] `Deployment`
- [x] `StatefulSet`
- [ ] `Job`
- [x] `Ingress`
- [x] `Pod`
- ~`Persistent Volume`~
- ~`Persistent Volume Claim`~

**Comment by hausdorff** [comment]
Critical resources are covered, we don't need to cover PVs and PVCs. Bumping the rest to M17.

**Comment by lblackstone** [comment]
Ingress is in progress in #283 
Job and StatefulSet will likely get bumped to M20

**Comment by krmayankk** [comment]
What is pulumi-kubernetes, from an initial thinking is it K8s api implemented by your own cloud in the way you seem fit ?

**Comment by hausdorff** [comment]
@krmayankk The appropriate venue to discuss that question is probably the [community slack](https://slack.pulumi.io/) or another issue with the label "question". If you'd like to follow up, please do so in one of those places.

But, because I can't resist answering questions, I'll provide a brief answer here. :) pulumi-kubernetes is the code that allows Pulumi to manage Kubernetes resources on an arbitrary Kubernetes cluster. Anywhere you have a kubeconfig file. It supplies logic for diffing, resource creation/update/deletion, as well as the logic for awaiting resource initialization.

**Comment by lblackstone** [comment]
Job is the only remaining Kind to implement for this issue. Punting until after #343 is complete.

**Comment by hausdorff** [comment]
Actually, let's close this in favor of #449, which tracks job completion.
